### PR TITLE
Implement Adtraction

### PR DESCRIPTION
**Pull request by gustafguner**

### CODE REVIEW

File: src/utils/tracking.ts [hunk 1]
```diff
@@ -43,3 +44,41 @@ export const { Track, TrackAction, Identify, IdentifyAction } = setupTrackers<
...
+  doEvent: () => void
+}
+
+export const adtraction = (
+  orderValue: number,
+  orderId: string,
+  emailAddress: string,
+  couponCode: string | null,
+) => {
+  // @ts-ignore
```
Comment (palmenhq) [inline src/utils/tracking.ts]: :( why?

File: src/utils/tracking.ts [hunk 2]
```diff
@@ -43,3 +44,41 @@ export const { Track, TrackAction, Identify, IdentifyAction } = setupTrackers<
...
+  const adt: Adt = ADT
+  adt.Tag = adt.Tag || {}
+  adt.Tag.t = 3
+  adt.Tag.c = 'SEK'
+  adt.Tag.tp = 1412601108
+  adt.Tag.am = orderValue
+  adt.Tag.ti = orderId
+  adt.Tag.xd = md5(emailAddress)
```
Comment (palmenhq) [inline src/utils/tracking.ts]: Are we trying to scrub personal data or do they explicitly request us to md5 hash the emails?

File: src/server/page.tsx [hunk 3]
```diff
@@ -60,6 +60,7 @@ const template = (
     Sentry.init(${JSON.stringify(sentryConfig())})
   </script>
   <script key="segment-snippet" nonce="${cspNonce}">${segmentSnippet}</script>
+  <script defer src="https://adtr.io/jsTag?ap=1412531808"></script>
```
Comment (palmenhq) [inline src/server/page.tsx]: this (and any other domains they might use) must be placed in the CSP headers

File: src/pages/Sign/sections/SignSubscription.tsx [hunk 4]
```diff
@@ -188,16 +189,45 @@ const StateComponent: React.SFC<StateComponentProps> = ({
...
-              />
-            )}
-          </CurrentLanguage>
-        )
+
+        if (process.env.NODE_ENV !== 'test') {
+          return (
+            <OfferContainer>
+              {(offer) => {
+                adtraction(
```
Comment (palmenhq) [inline src/pages/Sign/sections/SignSubscription.tsx]: this part bugs me. Why are we doing this on every render? Is that really what was intended?

File: src/utils/tracking.ts (hunk 2 above)
Comment (gustafguner) [reply src/utils/tracking.ts]: They explicitly request the email addresses to be MD5-hashed...

File: src/utils/tracking.ts (hunk 1 above)
Comment (gustafguner) [reply src/utils/tracking.ts]: I guess I could declare my own types for it, that's true :-)

File: src/server/page.tsx (hunk 3 above)
Comment (gustafguner) [reply src/server/page.tsx]: Good catch! 💯

**Pull request merged**